Close 'More', 'Edition Picker' and 'Account' menus with ESC key press

GHaberis opened the pull request:
## What does this change?

Accessibility fix.

Currently If a user opens the  'More' or 'Edition Picker' menus and tabs into them there's no way of closing the menus with their keyboard.

This PR fixes that by adding the ability to close the menus using the `ESC` key, which is recommended behaviour. This functionality is enabled by adding a `keyup` event listener to the menu.

I've also simplified the logic from https://github.com/guardian/frontend/pull/19631, as I found I could reuse the `innerHTML` from the `label`'s  replaced by `button`s. Doing this meant I could remove the `HTML` in `new-header.js` and also remove redundant styles from `_main-menu-toggle.scss`.

## What is the value of this and can you measure success?

You can now close the 'More' and 'Edition Picker' menus with your keyboard

## Does this affect other platforms - Amp, Apps, etc?

No

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

N/A

## Screenshots

N/A

## Tested in CODE?

~No~ Yes

PRBuilds commented:
PRbuilds results:


**Screenshots**
[wide.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/wide.png) • [desktop.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/desktop.png) • [tablet.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/tablet.png) • [mobile.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/mobile.png)



&#x1F49A; **Exceptions**
[thrown-exceptions.js](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/thrown-exceptions.js)





&#x1F49A; **A11y validation**
[a11y-report.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/a11y-report.txt)



&#x1F49A; **Microdata Validation**
[microdata.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/microdata.txt)

    

**Apache Benchmark Load Testing**
[loadtesting.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/loadtesting.txt)



**LightHouse Reporting**
[lighthouse.html](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19648/lighthouse.html)


    
--automated message

GHaberis commented on static/src/javascripts/projects/common/modules/navigation/new-header.js:
```diff
@@ -287,20 +287,23 @@ const toggleEditionPicker = (): void => {
-            `<span class="u-h">current edition: </span>${displayName}`,
-        clickHandler: toggleEditionPicker,
+    [EDITION_PICKER_TOGGLE_CLASS]: (event: Event): void => {
+        if (event.keyCode === ESC_KEY_ID) {
```
OK I'll get this updated 👍

sndrs commented:
does this need some removal of listeners?

paperboyo commented:
💎! (obviously)

GHaberis commented:
I've updated this PR so it could do with a re-review.

The main update is I've added logic that unregisters the `clickstream:click` listeners when a menu is closed using the `ESC` key.

This was required because if we don't do this when a user closes a menu with the `ESC` key, the next time the user clicks anywhere on the page the `clickstream:click` handler would execute and toggling the menu visible.
